fed/mosaic-output/src/main/java/org/eclipse/mosaic/fed/output/generator/websocket/WebsocketVisualizerServer.java

```diff
@@ -142,6 +134,46 @@ private VehicleUpdates reduceVehicleUpdates(VehicleUpdates original) {
         return new VehicleUpdates(original.getTime(), Collections.EMPTY_LIST, reducedUpdates, Collections.EMPTY_LIST);
     }
 
+    private void sendAgentUpdates(WebSocket socket) {
+        if (agentUpdatesReference.get() != null && !agentUpdatesReference.get().getUpdated().isEmpty()) {
+            AgentUpdates reduced = reduceAgentUpdates(agentUpdatesReference.get());
+            JsonElement jsonElement = new Gson().toJsonTree(reduced);
+            JsonObject jsonObject = new JsonObject();
+            jsonObject.add(AgentUpdates.TYPE_ID, jsonElement);
+            socket.send(jsonObject.toString());
+        }
+    }
+
+    private AgentUpdates reduceAgentUpdates(AgentUpdates original) {
+        List<AgentData> reducedUpdates = new ArrayList<>();
+        for (AgentData agent : original.getUpdated()) {
+            reducedUpdates.add(new AgentData(
+                    agent.getTime(), agent.getName(), agent.getPosition(), null, null, agent.getState()
```

Comment:
IMO this is better readable
```suggestion
            reducedUpdates.add(
                    new AgentData(agent.getTime(), agent.getName(), agent.getPosition(), null, null, agent.getState())
            );
```

Reply:
Quite subjective, but I can live with that.